Add classname and styles to Voronoi cells

sqren opened the pull request:
Closes #684

This change makes it possible to conditionally add classnames and styles to Voronoi cell, depending on the data points.

This PR also adds initial docs for the Voronoi diagram.

sqren commented on src/plot/voronoi.js:
```diff
@@ -28,7 +29,7 @@ function Voronoi({
       {voronoiInstance.polygons(nodes).map((d, i) => (
         <path
-          className="rv-voronoi__cell"
+          className={`rv-voronoi__cell ${cellClassName(d.data)}`}
```
I'm not a huge fan of the name `cellClassName` and perhaps it should also allow strings (not just functions). But I didn't want to make it more complex.

mcnuttandrew commented on src/plot/voronoi.js:
```diff
@@ -56,7 +57,8 @@ Voronoi.defaultProps = {
   onMouseUp: NOOP,
   x: d => d.x,
-  y: d => d.y
+  y: d => d.y,
```
might be better to have the default be something like:

```
cellClassName: d => d.className || ''
```

sqren replied:
Where would `d.className` come from? I can see that `d` is a polygon node from the voronoi instance but not clear on how classnames are attached to those.

sqren replied:
I see now that it comes from `d.data.className`, which is passed to the Voronoi as `nodes`.That makes me think, that `cellClassName` is unnecessary. I'll update this PR.

mcnuttandrew commented on src/plot/voronoi.js:
```diff
@@ -71,7 +73,8 @@ Voronoi.propTypes = {
   x: PropTypes.func,
-  y: PropTypes.func
+  y: PropTypes.func,
+  cellClassName: PropTypes.func
```
alphabetize here and above

sqren commented:
@mcnuttandrew I've updated the code and added a test for the Voronoi diagram. 
Starting from scratch I found it a little hard to add docs (not my strong suit). But I added a base, that will hopefully encourage others to improve upon it. I also wasn't clear on how to test the docs. `npm start` only gives me the show-case items. So there might be some glitches that I didn't catch.

Let me know what you think.

mcnuttandrew commented on src/plot/voronoi.js:
```diff
@@ -28,7 +28,7 @@ function Voronoi({
       {voronoiInstance.polygons(nodes).map((d, i) => (
         <path
-          className="rv-voronoi__cell"
+          className={`rv-voronoi__cell ${d.data.className || ''}`}
```
just for paranoia maybe do d.data && d.data.className

sqren commented:
@mcnuttandrew Good to hear :) Commited the last fixes.

sqren commented:
@mcnuttandrew Any idea when this will be published on npm? Thanks!

mcnuttandrew commented:
@balthazar ?

balthazar commented:
`1.8.1`
